test/helpers/position_assertions.h

```diff
@@ -178,6 +181,14 @@ class TypeAssertion final : public RangeAssertion {
     std::string toString() const override;
 };
 
+// does assertions using positions specified by # ^^^ usage: symbol
```

Comment:
It's confusing to have a class named `HighlightAssertion` when there's no `highlight:` assertion.

Could we add it to `UsageAssertion` instead (maybe `checkHighlight`)?